edx_oauth_client/middleware.py

```diff
@@ -69,69 +98,32 @@ def process_request(self, request):
-            'request_certificate',)
-
-        handle_local_urls = (
-            'i18n', 'search', 'verify_student', 'certificates', 'jsi18n', 'course_modes', '404', '500', 'i18n.js',
-            'wiki', 'notify', 'courses', 'xblock', 'change_setting', 'account', 'notification_prefs', 'admin',
-            'survey', 'event', 'instructor_task_status', 'edinsights_service', 'openassessment', 'instructor_report',
-            'logout'
-        )
-
-        handle_local_urls += auth_process_urls + api_urls
-        is_auth = request.user.is_authenticated()
+        start_url_path, _, _ = urlparse(current_url).path.strip("/").partition("/")
 
+        available_urls = API_URLS + LOCAL_URLS + OAUTH_PROCESS_URLS
         if settings.DEBUG:
-            debug_handle_local_urls = ('debug', settings.STATIC_URL, settings.MEDIA_URL)
-            handle_local_urls += debug_handle_local_urls
-
-        if request.path in ("/dashboard/", "/dashboard"):
-            if is_auth and DASHBOARD_URL_PATH:
-                return redirect(os.path.join(PROVIDER_URL, DASHBOARD_URL_PATH))
-            else:
-                return redirect(PROVIDER_URL)
-
-        r_url = re.compile(r'^/courses/(.*)/about').match(current_url)
-        if r_url:
-            return redirect(
-                os.path.join(os.path.join(PROVIDER_URL))
-            )
-
-        is_courses_list_or_about_page = False
-        r = re.compile(r'^/courses/%s/about' % settings.COURSE_ID_PATTERN)
-
-        if r.match(current_url):
-            is_courses_list_or_about_page = True
-
-        if COURSES_LIST_URL_PATH and request.path in ("/courses/", "/courses"):
-            return redirect(os.path.join(PROVIDER_URL, COURSES_LIST_URL_PATH))
-
-        if request.path.startswith('/u/') or request.path in ("/account/settings/", "/account/settings"):
-            if is_auth and USER_ACCOUNT_URL_PATH:
-                return redirect(os.path.join(PROVIDER_URL, USER_ACCOUNT_URL_PATH))
-            else:
-                return redirect(PROVIDER_URL)
-
-        if start_url not in handle_local_urls or is_courses_list_or_about_page:
-            if start_url.split('?')[0] not in handle_local_urls:
-                provider_url = PROVIDER_URL.rstrip("/") + "/"
-                return redirect("%s%s" % (provider_url, current_url))
-
-        if not is_auth and start_url not in auth_process_urls and start_url not in api_urls:
+            debug_handle_local_urls = ("debug", settings.STATIC_URL, settings.MEDIA_URL)
+            available_urls += debug_handle_local_urls
+
+        if request.user.is_authenticated():
+            if dashboard_url_path and request.path.strip("/") in reverse("dashboard"):
+                return redirect(urljoin(provider_url, dashboard_url_path))
+
+            if courses_list_url_path and request.path.strip("/") in reverse("courses"):
+                return redirect(urljoin(provider_url, courses_list_url_path))
+
+            if (
+                    user_account_url_path
```

Comment:
too big indentation, please take a look on the open edx python code style https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/style_guides/python-guidelines.html#breaking-long-lines

Reply:
yeah. Added a new version of the complex `if`